### PR TITLE
Add a manual test for the platform test

### PR DESCRIPTION
**Pull request by ogoffart**
(using Qt)

### CODE REVIEW

File: api/cpp/tests/manual/platform_qt/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+add_executable(platform_qt main.cpp)
+target_link_libraries(platform_qt PRIVATE Slint::Slint Qt::Gui Qt::Widgets Qt::GuiPrivate)
```
Comment (tronical) [inline api/cpp/tests/manual/platform_qt/CMakeLists.txt:2]: I suppose widgets are not really needed, are they? I see that the C++ code creates a `QApplication`, but I think a `QGuiApplication` should also suffice.

**Review by tronical — APPROVED** [review]
Makes sense to me to have that as a manual test, and using Qt as host is convenience to reduce the amount of platform-dependent boiler plate.

I suggest to remove the widgets dependency though.

File: api/cpp/tests/manual/platform_qt/CMakeLists.txt (hunk 1 above)
Comment (ogoffart) [reply api/cpp/tests/manual/platform_qt/CMakeLists.txt:2]: The problem is that, by default, we're going to us the native style and that needs a QApplication and assert with a QGuiApplication

Comment (tronical) [reply api/cpp/tests/manual/platform_qt/CMakeLists.txt:2]: That's a compelling argument :)

**Comment by ogoffart** [comment]
on linux, the CI errors with
```
1: /home/runner/work/slint/cppbuild/api/cpp/tests/test_datastructures: error while loading shared libraries: ../libslint_cpp.so: cannot open shared object file: No such file or directory
``` 
and similar

I guess that's because i moved the tests to the tests directory.
But it doesn't fail for me locally.  I wonder what's going on.

**Review by tronical — APPROVED** [review]
Nice that it the change is so straight-forward :)

**Pull request merged**